src/library_pthread.js

```diff
@@ -404,6 +413,10 @@ var LibraryPThread = {
     {{{ makeSetValue('__num_logical_cores', 0, 'cores', 'i32') }}};
   },
 
+  emscripten_transfer_canvas: function() {
+    Module['canvasTransferRequested'] = true;
```

Comment:
I like how you do it, actually. Seems more convenient to set it on Module in some cases.